editor/frontend/src/TransformElementsTool.ts

```diff
@@ -323,20 +458,21 @@ export class RotateElementsTool extends TransformElementsTool {
     if (RotateAbout.Point === this.rotateAbout)
       return super.transformAgendaDynamics(transform, context);
 
-    if (undefined === this._elementAlignedBoxes || undefined === this._elementOrigins || this._elementAlignedBoxes.length !== this._elementOrigins.length)
+    if (undefined === this._graphicsProvider)
       return;
 
-    const builder = context.target.createGraphicBuilder(GraphicType.WorldDecoration, context.viewport);
-    builder.setSymbology(context.viewport.getContrastToBackgroundColor(), ColorDef.black, 1, LinePixels.HiddenLine);
+    for (const data of this._graphicsProvider.data) {
+      const rotatePoint = Point3d.create();

```

Comment:
maybe take this out of the loop? Same thing for rotateTrans.

Reply:
done.